kernel/src/fs/fs_resolver.rs

```diff
@@ -134,7 +137,12 @@ impl FsResolver {
         let parent = lookup_ctx
             .parent()
             .ok_or_else(|| Error::with_message(Errno::ENOENT, "parent not found"))?;
-        if !parent.mode()?.is_writable() {
+
+        if parent
+            .inode()
+            .check_permission(Permission::MAY_WRITE)
+            .is_err()
+        {
```

Comment:
Is this check redundant given that a similar check is going to be performed in `Dentry::new_fs_child` at L151?

Reply:
Yes, I forgot to make the change. Thank you for your nice review